### ISSUE TITLE
Support playing of UHD BDs

### ISSUE THREAD

**Issue by helgetelefonmann** [Taapat/enigma2-plugin-blurayplayer#16]
Currently this doesn't seem to work... If I open a folder with a UHD-BD (ripped with DeUHD), I get message "Fehler beim Lesen..." (Error while reading...) which sounds somewhat unspecific. The thumbnail shows the logo of the film though.
Tested on VU+ Solo 4K with Planet Earth II UHD (.m2ts files play correctly on PC).

**Comment by Taapat** [comment]
On what image do you use this plugin?

**Comment by helgetelefonmann** [comment]
openATV 6.0
The plugin is enigma2-plugin-extensions-blurayplayer - 1+git133+660aff9-r0 - Enigma2 plugin to play Blu-ray discs

**Comment by hipernet** [comment]
I have also same in Egami

**Comment by hipernet** [comment]
not working this plugin in UHD

**Comment by Taapat** [comment]
Or you can play some of the .m2ts files (try the largest) in the enigma2 movie player?

Does your bluray folder structure match the standard?
https://en.wikipedia.org/wiki/Blu-ray#Directory_and_file_structure
Pay attention to the existence of the PLAYLIST directory with mpls files.

**Comment by helgetelefonmann** [comment]
Yes, the plain .m2ts files themselves can be played.
The folder structure also seems to be OK, there is indeed a folder BDMV/PLAYLIST which contains 12 .mpls files numbered sequentially (starting with 00001.mpls). They are in some binary format but all start with MPLS0300 and seem to contain references to the .m2ts-Files (e.g. @00001M2TS.)

**Comment by Taapat** [comment]
Can you show log from enigma2?
In the log should be [BlurayPlayer] blurayinfo.getTitles: ... with an error.

**Comment by helgetelefonmann** [comment]
Sorry, my log doesn't show any entries... do I have to enable some kind of debugging for this?

**Comment by Taapat** [comment]
I do not know how it is on openATV 6.0, but I assume that you need to specify a greater enigma debug level: https://wiki.openpli.org/Tips_and_Tweaks#Debugging_Enigma

**Comment by helgetelefonmann** [comment]
So it seems to be the index ID...?

17:42:06.0107 skin.py:1110 readSkin  [SKIN] Parsing embedded skin <embedded-in-'BlurayMain'>
17:42:06.0114 skin.py:1349 readSkin  [SKIN] processing screen <embedded-in-'BlurayMain'>:
index_parse.c:188: index.bdmv failed signature match: expected INDX0100 got INDX0300
index_parse.c:188: index.bdmv failed signature match: expected INDX0100 got INDX0300
[blurayinfo] Failed to open:/media/hdd/discs/Kathia Zubin/
[blurayinfo] getTitles: error in parse!

**Comment by Taapat** [comment]
Looks like that.
What version of libbluray library is used in your image?
I see that accept version 0300 files added on Aug. 28, 2017: http://git.videolan.org/gitweb.cgi/libbluray.git/?p=libbluray.git;a=commitdiff;h=440d9a46de4ddf578870237c29b7f4d714da7a6f
and should work on the current 1.0.2 release.

**Comment by helgetelefonmann** [comment]
OpenATV 6.2 (which I have currently) uses libbluray - v0.9.3+git2490+efcde25-r0 - which is indeed quite old and doesn't seem to support this. Any idea on how I can get a newer one (besides compiling it myself)?

**Comment by Taapat** [comment]
What receiver with which architecture do you use?
I have already added a libbluray 1.0.2 libraries for cortexa15hf-neon-vfpv4 and armv7ahf-neon architecture in the plugin download folder.
But they come from openpli-oe-core, so there's no guarantee that they will work on an OpenATV.

**Comment by helgetelefonmann** [comment]
Installation seems to be no problem...
root@vusolo4k:~# opkg install /hdd/temp/libbluray_v1.0.2+git2704+8c15fda-r0.0_armv7ahf-neon.ipk
Upgrading libbluray from v0.9.3+git2490+efcde25-r0 to v1.0.2+git2704+8c15fda-r0.0 on root.
Installing libbluray (v1.0.2+git2704+8c15fda) on root.
Removing obsolete file /usr/lib/libbluray.so.1.10.0.
Removing obsolete file /usr/lib/libbluray.so.1.
Configuring libbluray.

**Comment by helgetelefonmann** [comment]
...but duh, the box crashes upon starting the plugin...

**Comment by helgetelefonmann** [comment]
First error claims libbluray.so.1 is missing - if I replace it with a link to the new libbluray, the plugin starts and I can open the UHD folder, but then I get a crash again:

16:26:44.1200 {   } skin.py:1110 readSkin [SKIN] Parsing embedded skin <embedded-in-'BlurayMain'>
16:26:44.1208 {   } skin.py:1349 readSkin [SKIN] processing screen <embedded-in-'BlurayMain'>:
16:26:44.1564 [   ] bsod.cpp:328 print_backtrace Backtrace:
16:26:44.1567 [   ] bsod.cpp:336 print_backtrace /usr/bin/enigma2(_Z17handleFatalSignaliP9siginfo_tPv) [0x8D5D8]
16:26:44.1568 [   ] bsod.cpp:336 print_backtrace /lib/libc.so.6(__default_rt_sa_restorer) [0xB60ECAD0]
16:26:44.1568 [   ] bsod.cpp:348 handleFatalSignal -------FATAL SIGNAL

**Comment by Taapat** [comment]
It seems to plugin need compile with this library support.
If you have py files in the plugin folder, you can try to remove pyo files and restart the enigma, but I think it will not help.
Normally would by, if you're a developer of your image ask update libbluray on feed.

**Comment by helgetelefonmann** [comment]
Hm, there seems to be some hard-coding in the OE bitbake recipe for libbluray (from you?) which leads to libbluray-0.9.3 still being used. After removing this and making a new OpenATV 6.1 image I get libbluray-1.0.2 now, but I still get the error mentioned above. But the reason seems to be different now:

[blurayinfo] Failed to open:/media/hdd/discs/Kathia Zubin/
[blurayinfo] getTitles: error in parse!
12:48:32.6211 Plugins/Extensions/BlurayPlayer/BlurayUi.py:203 OpenDisc  [BlurayPlayer] blurayinfo.getTitles: error return without exception set

Sounds somewhat unspecific, though.

**Comment by helgetelefonmann** [comment]
Any news in this issue?

**Comment by Taapat** [comment]
Do you use the plugin from your image feed?

**Comment by helgetelefonmann** [comment]
Yes, but changed it as described above to get the higher version you recommended. Unfortunately I have no idea why the bitbake recipe is modified in a way that only version 0.9.3 is used. After changing, 1.0.2 compiles fine without errors.

**Comment by noX1609** [comment]
i have the same problem.
I'm also using openATV 6.2 at an ax HD51.
Is there an update available for the libbluray or could someone help me compiling the update for my enignma2 receiver? Don't have experienced knowledge of linux so don't know how to update it via ssh, would be really happy for help :)

**Comment by Taapat** [comment]
Do I understand you correctly?
You are using a plugin from openATV 6.2 feed and also libbluray 1.0.2  from openATV 6.2 feed?
What kind of error is in log?

If you think that the plugin from feed uses an old libbluray, then you need to ask your developer to update plugin version after the libbluray update.
Normally build system should do it automatically, but if openATV does not use PR service, then it does not do without update the plugin version.

**Comment by noX1609** [comment]
No i use 6.2 from the feed and tried the 1.02 libbluray from the download linked at the readme of this plugin from here: http://taapat.ho.ua/enigma2-plugin-blurayplayer/). The arm Version installs fine but after that if i try to open the Bluray player and enigma2 shows a bluscreen and reboots the receiver.
On the feed i can only find 0.9.3 but with this version the playback / loading of the iso doesn't work with the same message as the other user get (mean this one: index_parse.c:188: index.bdmv failed signature match: expected INDX0100 got INDX0300).

So is there anyone out there who could compile the libbluray for enigma2 with support for the AX 4K HD51? Would really appreciate that because i don't know how to compile it for myself :(
I know a lot of software development but with xcode on OSX so until now i only compiled plugins with tutorials which told me what i need to do step by step.
If there is such a tutorial out there i would be also happy with that, then i would compile it for my self but without such a tutorial it could take me several weeks to get the plugin running^^ 

If that's important: I only have the problem when i try to load/play a UHD Bluray iso (I'ts a complete rip so everything that would be included on the disk is included there too.
Also tried extracting the iso but it leads to the same error so the problem is libbluray which doesn't support UHD Disks in 0.9.3
Also looked on the oTV feed but there is no later version than 0.9.3

So please help me because got a new tv with UHD support and want to play my UHD Disks but i don't own a Hardware bluray player except Xbox One and PS4 but they both don't support UHD, only Xbox One S and x and the new PS4 slim and pro support UHD :(

**Comment by helgetelefonmann** [comment]
I did just what you described (installed toolchain for compiling OpenATV 6.1 from scratch, changed BB recipe so not libbluray 0.9.3 but 1.0.2 gets compiled/installed) - but when trying to load a UHD BD I get the following error.

[blurayinfo] getTitles: error in parse!
12:48:32.6211 Plugins/Extensions/BlurayPlayer/BlurayUi.py:203 OpenDisc [BlurayPlayer] blurayinfo.getTitles: error return without exception set

**Comment by Taapat** [comment]
Yes, sounds somewhat unspecific, though.
Maybe you have the opportunity give the link to such UHD-BD.
When I have time I could add an extra log in blurayinfo.c code and find why there is such error.
